### ISSUE TITLE
[Verification] App still verify email by 6-digit OTP even changed to verify by login link

### ISSUE THREAD

**Issue by joyzoursky** [authgear/authgear-server#3326]
## Background
- Environment: Staging
- Testing app: refuse-anchor-618, qademoappnative
- Demo apps (native): iOS 1.0(1692936706), android 1.0(1692847081)
- Device: iPhone X (iOS 16.0), pixel 5a (android 13), iPhone SE (iOS 16.4.1 (a))

## Description
Admin user has changed the login method - email verification to "Login Link", but after save and configure on the app again, it still shows 6-digit OTP when user sign up new account.

## Steps to reproduce
1. Select a project on Authgear portal
2. Go to Login Methods > Passwordless > Verification
3. Select to Verify email by "Login Link"
4. Save
5. Configure the application on demo app
6. Click Authenticate to sign up
7. See that the next step is 6-digit OTP but not login link

## Expected result
Should change to use Login link for verification after user changed on portal.

## Screenshot
<img width="1552" alt="Screenshot 2023-08-29 at 4 19 46 PM" src="https://github.com/authgear/authgear-server/assets/12539486/a9c2d770-9807-40ed-8718-2ff6ac9945c2">

Demo app (after saving on portal and clicked "Configure" button again):

https://github.com/authgear/authgear-server/assets/12539486/1afe5044-a585-41e2-9b70-d1d44e017e28

**Comment by joyzoursky** [comment]
Todo:
- Add `Sign up and verify email by` dropdown
- Rename current dropdown to `Login by`

**Comment by joyzoursky** [comment]
@fungc-io @louischan-oursky Found that the field also appears on "Enter password" pages, what function does the "Verify email by" refer to on those pages? It should be verify but not Login?

It seems that the solution in https://github.com/authgear/authgear-server/issues/3168#issuecomment-1611455947 is still confusing after the change.

**Comment by louischan-oursky** [comment]
"Verify email by" always means "What does the user receive when they sign in"

**Comment by fungc-io** [comment]
Will this work:

- Add and support `Sign up and verify email by` dropdown

Is always there if users can login/signup by email.

- Rename current dropdown to `Login by`

If Email & Passwordless is enabled.

**Comment by fungc-io** [comment]
I think in the portal, the "Verify phone number by Whatsapp/SMS" actually means both 
- Login
- Signup and verify
Which is inconsistent with the email, which are separated.

@louischan-oursky Are signup/verify and login separated configs for Phone under the hood?

**Comment by louischan-oursky** [comment]
> Are signup/verify and login separated configs for Phone under the hood?

No, they are NOT separated. Your explanation is accurate. Whatsapp/SMS is applied to Phone signup/verify/login, while Code/LoginLink is applied to Email login only.

**Comment by fungc-io** [comment]
**In email:**

- Add and support `Sign up and verify email` by dropdown

Is always there if users can login/signup by email.

- Rename current dropdown to `Login by`

If Email & Passwordless is enabled.

**In Phone**

Rename current dropdown to 
- If passwordless is enabled: `Sign up, verify phone and Login by`
- If it's disabled: `Sign up, verify phone by`
